### DISCUSSION TITLE
[auth][error] TypeError: fetch failed

### DISCUSSION THREAD

**Discussion by danisal** [nextauthjs/next-auth#11443] [Help]
Hi,

I'm facing an issue for which I just found the solution (?), but nevertheless, I wanted to ask for your help since this doesn't seem like a solution to me.

I'm using the latest version of next-auth (5.0.0-beta.19) with the provider Zitadel, and this is the screen I see when I try to deploy the next app on Vercel with an auth proxy also deployed on Vercel.

![image](https://github.com/user-attachments/assets/feea3c5b-d2cc-4398-a3f8-9dc1e4ccfa68)

And the server console only tells me this:

```shell
[auth][error] TypeError: fetch failed
    at node:internal/deps/undici/undici:12502:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getAuthorizationUrl (webpack-internal:///(rsc)/../../node_modules/next-auth/node_modules/@auth/core/lib/actions/signin/authorization-url.js:24:35)
    at async Module.signIn (webpack-internal:///(rsc)/../../node_modules/next-auth/node_modules/@auth/core/lib/actions/signin/index.js:16:56)
    at async AuthInternal (webpack-internal:///(rsc)/../../node_modules/next-auth/node_modules/@auth/core/lib/index.js:72:24)
    at async Auth (webpack-internal:///(rsc)/../../node_modules/next-auth/node_modules/@auth/core/index.js:126:34)
```

Using the same locally, with a local instance of the auth-proxy, everything works just fine.

[By chance](https://www.reddit.com/r/react/comments/1alvtjo/fetch_not_working_in_server_component_giving/), I've found that disabling the `NODE_TLS_REJECT_UNAUTHORIZED` everything works just fine...

I don't know why this happens, but it isn't a good solution for me.

Does anyone know why this happens? Since everything is deployed on Vercel, it should work, I would say

**Comment by danisal** [comment]
For reference, let me say that this happened when I tried to log in. The process is supposed to bring the user to the Zitadel login screen, but it doesn't even reach Zitadel; this happens as soon as I click the login button.

**Reply by aspirin2d** [reply]
Same error here...

**Reply by MiguelAngelOne** [reply]
yeah same error here with zitadel even with NODE_TLS_REJECT_UNAUTHORIZED to zero, I'm deploying in fly, somebody had any luck resolving this issue?

**Reply by cshaa** [reply]
It's not enough to put `NODE_TLS_REJECT_UNAUTHORIZED=0` into .env, I had to either start the demo app using `NODE_TLS_REJECT_UNAUTHORIZED=0 make start`, or put it directly into the makefile like this:
```Makefile
.PHONY: start

start:
	npm install
	NODE_TLS_REJECT_UNAUTHORIZED=0 npm start

```

**Comment by wujibear** [comment]
I'm also experiencing this.
Not sure if there's something wrong with my config somewhere, or what this is meant to mean.

I tried setting `NODE_TLS_REJECT_UNAUTHORIZED` to zero, and then to false. I did that just to see if there's a similarity there for me.
Neither bypassed whatever this issue is, but I get this same error message.

**Comment by 985563349** [comment]
I'm having the same problem with google authentication

**Reply by imrbrad** [reply]
Me too

**Reply by 985563349** [reply]
I have solved this problem. The reason is that my nodejs can't access google services.

**Reply by danalex7212** [reply]
what did you do to give nodejs access to google services  ? I am facing this issue with microsoft .

**Reply by 985563349** [reply]
> what did you do to give nodejs access to google services  ? I am facing this issue with microsoft . 

I can't access the Google service in my area, so I used the tun proxy mode.

**Reply by Larusyang** [reply]
I just meet the same problem . Github is ok, but Google....

**Reply by ajhous44** [reply]
@985563349 What is "Tun proxy mode"? 

Has anyone else resolved this? Specifically with entra id?

**Reply by 985563349** [reply]
@ajhous44 proxy for all OS traffic.

**Reply by emexnord** [reply]
on my side, I forgot to add my endpoint to the Authorized JavaScript Origins when configuring it in Google. after fixing that, it worked fine

**Reply by Rhys97W** [reply]
> @985563349 What is "Tun proxy mode"?
> 
> Has anyone else resolved this? Specifically with entra id?

Did you solve this? I have the same with Entra

**Comment by 805131398** [comment]
不会是因为开了 VPN 的原因把? 我也是 Github 可以, google 不行。

**Reply by Mierku** [reply]
这报错就是国内连不上google服务，本地开发得开vpn然后配置node的流量分发

**Reply by mangoyi** [reply]
code有吗？大佬，我尝试加代理了不行。

**Reply by Mierku** [reply]
代码有点复杂 后面我发代理换Tun模式 一样能行

**Comment by otakutyrant** [comment]
As a Chinese I encountered this error, blame GFW. I enabled System Proxy and TUN mode in Clash Verge to solve it. So this issue is almost network issue.
